envoy/stats/stats.h

```diff
@@ -129,10 +130,12 @@ using CounterSharedPtr = RefcountPtr<Counter>;
  */
 class Gauge : public Metric {
 public:
+  // TODO(diazalan): Rename ImportMode to more generic name
   enum class ImportMode {
     Uninitialized, // Gauge was discovered during hot-restart transfer.
     NeverImport,   // On hot-restart, each process starts with gauge at 0.
     Accumulate,    // Transfers gauge state on hot-restart.
+    Hidden,        // For deferred-stats gauges
```

Comment:
I think we should specify here what 'hidden' means. It's fine to give the use-case also. But we should specify:
 * will be transferred on hot-restart
 * will not show up in stats-sinks and admin. Though as I've mentioned it might be helpful when reasoning about behavior to make /stats?hidden work -- maybe calling it that the stats endpoint should *only* show hidden stats and nothing else when enabled.

Speaking of which, are we going to include the 'hidden' behavior described here in this PR? I think it would probably make sense. Right now the only effect of 'hidden' is to block disabling of those stats.

Reply:
I was thinking the hidden behavior would be included in a follow up PR